test/test_search.rb

```diff
@@ -0,0 +1,96 @@
+# ~*~ encoding: utf-8 ~*~
+path = File.join(File.dirname(__FILE__), 'helper')
+require File.expand_path(path)
+
+context 'Wiki search' do
+
+  SEARCH_TEST_PAGE = <<EOF
+Hobbits are small. Hobbits live in Hobbiton.
+One thing Hobbits do not like is Orcs.
+Something a hobbit does like is pastry.
+Some other words.
+EOF
+
+  SEARCH_TEST_LINES = SEARCH_TEST_PAGE.split(/[\n\r]+/)
+
+  setup do
+    @path = cloned_testpath('examples/lotr.git')
+    @wiki = Gollum::Wiki.new(@path, :page_file_dir => 'Rivendell')
+    @wiki.write_page('bar', :markdown, 'bar', commit_details)
+    @wiki.write_page('filename:with:colons', :markdown, '# Filename with colons', commit_details)
+    @wiki.write_page('foo', :markdown, "# File with query in contents and filename\nfoo", commit_details)
+    @wiki.write_page('Hobbit Info', :markdown, SEARCH_TEST_PAGE, commit_details)
+  end
+  
+  test 'search results should be able to return a filename with an embedded colon' do
+    results = @wiki.search('colons')
+    assert_not_nil results
+    assert_equal 'filename:with:colons.md', results.first[:name]
+    assert_equal 2, results.first[:count]
+  end
+
+  test 'search results should make the content/filename search additive' do
+    # There is a file that contains the word 'foo' and is called 'foo', so it should
+    # have a count of 2, not 1...
+    results = @wiki.search('foo')
+    assert_equal 2, results.first[:count]
+  end
+
+  test 'search results should not include files that do not match the query' do
+    results = @wiki.search('foo')
+    assert_equal 1, results.size
```

Comment:
I guess you should also explicitly assert that a certain file is not present, given the name of this test.